File: frontend/src/app/core/types/analytics.ts
export interface ApplicationDataPoint {
  day: string;
  sent: number;
  responses: number;
}

export interface PlatformDataPoint {
  platform: string;
  applied: number;
  responses: number;
}

export interface AnalyticsData {
  applicationData: ApplicationDataPoint[];
  platformData: PlatformDataPoint[];
}

File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class ApplicationDataPoint(BaseModel):
    day: str
    sent: int
    responses: int


class PlatformDataPoint(BaseModel):
    platform: str
    applied: int
    responses: int


class AnalyticsResponse(BaseModel):
    application_data: list[ApplicationDataPoint]
    platform_data: list[PlatformDataPoint]
